lib/manage.js

```diff
@@ -165,22 +166,80 @@ export async function manage({meeting, meetings, xmppOptions}) {
+          if(msg && msg.event == 'SPEECH' && !msg.is_interim) {
+            const commenter = msg.participant.name.replace(/ /g, '_');
+            const rawMessage = msg.transcript[0].text + '.';
+            const message = commenter + ': ' +
+              rawMessage[0].toUpperCase() + rawMessage.slice(1);
+            if(ircOptions.log) {
+              log({logFileBasePath, nick, message});
+            }
+            ircSay(`${nick}: ${message}`);
+            await xmppSay({info: `${nick}: ${message}`, meeting, xmppClient});
+          }
+
+          // the message is a new poll message
+          if(msg && msg.type === 'new-poll') {
+            if(!Array.isArray(msg.answers)) {
+              console.error('Poll with non-array answers');
+              return;
+            }
+            const pollNum = ++polls.count;
+            polls.byId[msg.pollId] = {num: pollNum, msg, votes: {}};
+            polls.idsByNum[pollNum] = msg.pollId;
+            const numOptions = msg.answers.length;
+            // If the poll is created from IRC, the participant/nick is the
+            // bot, so use senderName instead.
+            const bridged = stanza.attrs.from.endsWith('/cgbot');
+            const name = bridged ? msg.senderName : nick;
+            const message = `${name} created poll #${pollNum} with ${numOptions} options.`;
+            ircSay(message);
+            if(ircOptions.log) {
+              log({logFileBasePath, nick: ircOptions.nick, message});
+            }
+            const message2 = `... Question: ${msg.question}`;
+            ircSay(message2);
+            if(ircOptions.log) {
+              log({logFileBasePath, nick: ircOptions.nick, message: message2});
+            }
+            msg.answers.forEach(async (option, i) => {
+              const message = `... Option ${i+1}: ${option}`;
+              ircSay(message);
               if(ircOptions.log) {
-                log({logFileBasePath, nick, message});
+                log({logFileBasePath, nick: ircOptions.nick, message});
               }
-              ircSay(`${nick}: ${message}`);
-              await xmppSay({info: `${nick}: ${message}`, meeting, xmppClient});
+            });
+          }
+
+          // the message is a poll answer message
+          if(msg && msg.type === 'answer-poll') {
+            const poll = polls.byId[msg.pollId];
+            if(!poll) {
+              console.error('Unknown poll ' + msg.pollId);
+              return;
+            }
+            const answers = poll.msg.answers;
+            poll.votes[msg.voterId] = msg;
+            const message = `${msg.voterName} answered poll #${poll.num}: `
+              + msg.answers.map((chosen, i) => chosen
+                ? `(${i+1}) ${answers[i]}` : '')
+                .filter(Boolean).join(', ');
+            ircSay(message);
+            if(ircOptions.log) {
+              log({logFileBasePath, nick: ircOptions.nick, message});
```

Comment:
Is there a "finalize poll" or "close poll" command that shows the totals? It might be difficult for more complex polls to understand the final tally? I know we have the "PROPOSAL / RESOLUTION" syntax already for proposals... but POLLs are different, maybe?

Reply:
The `votes` IRC command added in this PR shows the votes summary for a poll (i.e. the latest poll, or a previous one if referenced); it aims to present similar information as the web view in Jitsi (i.e. number of votes per option, names of voters, and percentage of voters who chose each option). In the web UI, the number of voters is shown in the "details" view but in this IRC implementation I have it show every time - If that is excessive with many voters, it could be adjusted.

There's no operation in the Jitsi front-end to close, remove or finalize a poll. New polls can be created and the old ones remain. The closest I can see that we could do to close a poll purely in the bot, I think would be the following:
- The bot could state in IRC and Jitsi chat that the poll is to be considered closed.
- The bot could zero out the poll - removing all existing votes. I think this could work because the web UI and conference system does not appear to verify that a vote message's sender property corresponds to the actual message sender; also, even though a user cannot remove their vote in the web UI (the client requires at least one option chosen), I don't think that is enforced internally.
- The bot could detect and revert votes made on a poll that has been marked closed. Additionally it could send a private message to the voter as a warning/alert.

Polls are different from proposal/resolutions but might be similarly useful in some situations.
For example, to emulate the current proposal method of evaluating a proposed resolution, a poll could have the question be a proposal, and options "+1", "0", and "-1" (or "yes", "no", "abstain", "block", "still deciding").

If there are multiple alternatives to be considered as proposals, a single poll could have a more general question, and give the proposed alternatives as options. Participants can vote for more than one option in a poll. This could be useful for comparing multiple possible alternatives. This might not have the nuance of the earlier method, as each option would get only a binary response per person (or ternary if you consider not voting as a response) rather than a spectrum of responses. It could be followed up with a poll specifically for an option to consider further.